### PR TITLE
Homework + Extra Credit 

### PR DESCRIPTION
**Pull request by johnnygoodman**
I changed a lot of stuff up. The big decision was passing fighter information to the Turn class. I kept trying to work around the winning moves and winning fighters being decoupled and no matter how I did it, it wouldn't make sense. I wanted a SQL bridge table....

I also didn't like the "output the moves/score for each round and then the overall". I wanted to see which fighter won each round. I changed the winner method in the Match class to a score method, then passed back a hash instead of just an opponent back. 

Here is sample script output:

$ ruby superfight.rb 
What is your first fighter's name?
Ali
What is your second fighter's name?
Gee

Round 1
Winner: Gee (strike, 78)
Loser: Ali (strike, 23)

Round 2
Winner: Ali (strike, 48)
Loser: Gee (strike, 8)

Round 3
Winner: Ali (strike, 81)
Loser: Gee (strike, 41)

Round 4
Winner: Gee (strike, 20)
Loser: Ali (strike, 1)

Round 5
Winner: Gee (strike, 84)
Loser: Ali (strike, 11)

Round 6
Winner: Ali (strike, 93)
Loser: Gee (strike, 42)

Round 7
Winner: Ali (strike, 96)
Loser: Gee (strike, 25)

Round 8
Winner: Ali (strike, 69)
Loser: Gee (strike, 44)

Round 9
Winner: Ali (strike, 55)
Loser: Gee (strike, 53)

Round 10
Winner: Ali (strike, 89)
Loser: Gee (strike, 71)

Round 11
Winner: Ali (strike, 99)
Loser: Gee (strike, 47)

Round 12
Winner: Ali (strike, 94)
Loser: Gee (strike, 55)

Round 13
Winner: Ali (strike, 82)
Loser: Gee (strike, 74)

Final Match Results: Ali defeats Gee (10 to 3)

I'd love some feedback. I'm interested in both general correctness and style tips. 

P.S. rand(100) returns a value from 0 to 99. rand(100) + 1 returns a value from 1 to 100. I had my rake tests randomly error out and was pretty confused for a while there.

### CODE REVIEW

**Comment by johnnygoodman** [comment]
The code above is 1 and 2 on the extra credit. I'm having trouble with 3, the block/block retry.

[I found this on stack](http://stackoverflow.com/questions/4609872/rspec-commandline-variable-input) but am having trouble refactoring testing like this: 

```
it "should have 13 turns" do
    subject.turns.count.should eq(13)
end
```

It gets hung on turn 1, because turn 1 now required prompts.

File: lib/fighter.rb [hunk 1]
```diff
@@ -1,10 +1,29 @@
 require_relative "move"
 class Fighter
 
-	attr_reader :strike, :block, :name
+	attr_reader :name, :strike, :block, :name
```
Comment (jwo) [inline lib/fighter.rb:4]: Don't need to have name here twice.

File: spec/fighter_spec.rb [hunk 2]
```diff
@@ -14,4 +14,11 @@
 	it "should have a name" do
 		Fighter.new("Mike Tyson").name.should eq("Mike Tyson")
 	end
+	
+	it "should be type strike if user chooses strike" do
+    hrm = subject.choose_move_type
+    hrm.stub!(:gets){ "1\n" }
+    hrm.stub!(:gets){ "2\n" }
```
Comment (jwo) [inline spec/fighter_spec.rb:21]: Were you trying to test user input here and had trouble?

**Comment by jwo** [comment]
Overall looks solid. My only recommendation is to not use `gets` inside of the fighter class -- That's sort of poking a big hole in your app and letting the user-interface deep down inside it... Crossing a barrier.

Instead, we could have the Match object not do a 13.times thing, but instead receive commands itself. So the Match object would have a `next_turn!` method that received [:strike, :block] or similar, and builds the turn then. Then the UI layer would do the looping, gets'ing and such.

**Pull request closed without merging**